### PR TITLE
Added xsoar engine related functions to the CSP

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Added to the CSP the following two functions:
1. The `is_integration_instance_running_on_engine()` function which determines whether the current integration instance runs on an xsoar engine.
2.  The `get_engine_base_url()` function which gets an engine ID and returns its base URL.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShacharKidor** [comment]
@kobymeir @JasBeilin - FYI - the new functions were tested on XSOAR 6, 8, and XSIAM.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -8596,12 +8596,86 @@ def is_xsiam():
...
 
+def is_integration_instance_running_on_engine():
+    """Determines whether the current integration instance runs on an xsoar engine.
+    If yes - returns the engine id.
+
+    :return: The engine id iff the instance is running on an xsaor engine.
+    :rtype: ``str``
+    """
+    engine_id = ''
+    integrations_raw_response = demisto.internalHttpRequest(
```
Comment (kobymeir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8620]: i've seen some places call this function with a size:1000.
we need to also do it here. or to properly paginate over the results.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -8596,12 +8596,86 @@ def is_xsiam():
...
+    except json.JSONDecodeError:  # type: ignore[attr-defined]
+        demisto.debug('Unable to load response {}'.format(integrations_body_raw_response))
+        integrations_body_response = {}
+
+    instances = integrations_body_response.get('instances', [])
+    instance_name = demisto.integrationInstance()
+    demisto.debug("Search for the data of the {} instance.".format(instance_name))
+    for instance in instances:
+        if instance_name == instance.get('name', ''):
+            engine_id = instance.get('engine', '')
```
Comment (kobymeir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8635]: add `break`?
or change to a next([... list comp...])

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -8596,12 +8596,86 @@ def is_xsiam():
...
+
+    if engine_id:  # engine_id = '' for instances that don't run on engine
+        demisto.debug("The {} instance runs on an xsoar engine, engine ID is: {}".format(instance_name, engine_id))
+        return engine_id
+    else:
```
Comment (kobymeir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8640]: no-else-on-return

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 4]
```diff
@@ -8596,12 +8596,86 @@ def is_xsiam():
...
+        demisto.debug("The {} instance does not run on an xsoar engine.".format(instance_name))
+        return ''
+
+
+def get_engine_base_url(engine_id):
+    """Gets the xsoar engine id and returns it's base url. 
+    For example: for engine_id = '4c80ce87-5a73-401c-b6a7-f4f12f86ff32', base url = '11.180.111.111:1443'.
```
Comment (kobymeir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8647]: the UUID number seems not mocked...

**Review by kobymeir — APPROVED** [review]

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 2 above)
Comment (ShacharKidor) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8635]: Fixed - Thanks!

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 3 above)
Comment (ShacharKidor) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8640]: Fixed - Thanks!

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 4 above)
Comment (ShacharKidor) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8647]: It is an ID not a password, and the engine is from a Tests tenant so it is not a secret... but I'll mock it.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py (hunk 1 above)
Comment (ShacharKidor) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8620]: Added - Thanks.

**Comment by kobymeir** [comment]
### This PR was automatically updated by a [GitHub Action](https://github.com/demisto/content/actions/runs/12122003002)
- **Base** pack version was bumped to **1.36.2**.

To stop automatic version bumps, add the `ignore-auto-bump-version` label to the github PR.

**Pull request merged**